File: scripts/example_ping.lua
-- Example Lua script for netcheck
-- This script demonstrates how to write a basic check
-- The 'hostname' variable is automatically provided by netcheck

-- Simple example: always succeeds for demonstration
-- In a real script, you would perform actual checks here

-- You can use os.execute() to run system commands
-- For example, let's do a simple ping check

local ping_cmd
if package.config:sub(1,1) == '\\' then
    -- Windows
    ping_cmd = string.format('ping -n 1 -w 2000 %s', hostname)
else
    -- Unix/Linux/macOS
    ping_cmd = string.format('ping -c 1 -W 2 %s > /dev/null 2>&1', hostname)
end

local exit_code = os.execute(ping_cmd)

-- Set result to true if ping succeeded, false otherwise
if exit_code == 0 or exit_code == true then
    result = true
else
    result = false
    error_message = string.format("Ping to %s failed", hostname)
end

File: scripts/example_ping.py
#!/usr/bin/env python3
"""
Example Python script for netcheck
This script demonstrates how to write a basic check in Python
"""

import sys
import subprocess
import platform

def main():
    # Check if hostname argument was provided
    if len(sys.argv) < 2:
        print("Error: No hostname provided", file=sys.stderr)
        sys.exit(1)

    hostname = sys.argv[1]

    # Perform a ping check
    # Adjust ping command based on platform
    if platform.system().lower() == "windows":
        # Windows: ping -n 1 -w 2000 hostname
        cmd = ["ping", "-n", "1", "-w", "2000", hostname]
    else:
        # Unix/Linux/macOS: ping -c 1 -W 2 hostname
        cmd = ["ping", "-c", "1", "-W", "2", hostname]

    try:
        # Run ping command
        result = subprocess.run(cmd, capture_output=True, timeout=5)

        if result.returncode == 0:
            # Ping successful
            sys.exit(0)
        else:
            # Ping failed
            print(f"Ping to {hostname} failed", file=sys.stderr)
            sys.exit(1)
    except subprocess.TimeoutExpired:
        print(f"Ping to {hostname} timed out", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()
